### PR TITLE
Messages

### PR DESCRIPTION
**Pull request by alshakerM**
Added messages and buttons to the icons on the navBar to access said messages.
Also changed the icons color when they are clicked.

### CODE REVIEW

**Comment by alshakerM** [comment]
Fixes a bug that happens when you click on one of the users that makes the icon color changes back to white.

**Comment by alshakerM** [comment]
Deleted unused imports

File: src/Messages/Messages.css [hunk 1]
```diff
@@ -0,0 +1,26 @@
+.messageText {
+  border: 1px solid #efefef;
+  border-radius: 22px;
+  max-width: 236px;
+  min-height: 44px;
+  padding: 16px;
+  justify-self: center;
+  align-self: center;
+  text-align: center;
```
Comment (alshakero) [inline src/Messages/Messages.css:9]: These are unneeded since it's not a flex.

File: src/Messages/Messages.css [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  box-sizing: border-box;
+}
+
+.messageText.is-sent {
+  background: #efefef;
+  display: flex;
+  justify-content: flex-end;
+  margin-right: 20px;
```
Comment (alshakero) [inline src/Messages/Messages.css:17]: You only need the background color and the margin.

File: src/Messages/Messages.css [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+.messages-user-thumbnail-message-text {
+  display: flex;
+  align-items: center;
+  gap: 10px;
+}
+.messages-user-thumbnail-message-text.is-sent {
+  align-self: flex-end;
+}
```
Comment (alshakero) [inline src/Messages/Messages.css:26]: You can add this file to the CSS module.

File: src/Messages/Messages.js [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+  whenWasActive,
+  yourMessages,
+} from './Messages.module.css';
+
+function UserSection() {
+  return (
+    <div className={userSection}>
+      <div className={usernameIconSection}>
+        <p>
+          <strong>7kami_me</strong>{' '}
```
Comment (alshakero) [inline src/Messages/Messages.js:44]:
```suggestion
          <strong>7kami_me</strong>
```

File: src/Messages/Messages.js [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+          <Link to={`/direct/t/${user.from_user_id}`}>
+            <img
+              className={SenderThumbnail}
+              src={user.from_user_thumbnail}
+              alt="sender avatar"
+            />
+          </Link>
+          <div>
+            <p className={senderUsername}>{user.from_username}</p>
+            <p className={hasSentMessage}>Active 20min ago</p>
```
Comment (alshakero) [inline src/Messages/Messages.js:90]: Why lock it to 20 mins ago? It should be the relative time of the last message.

File: src/Messages/Messages.js [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+  const userIndex = messagesData?.findIndex(
+    (user) => user.from_user_id === fromUserId
+  );
+
+  const date = new Date(
+    messagesData[userIndex]?.messages[
+      messagesData[userIndex]?.messages.length - 1
+    ].sent_on
+  );
+  console.log(fromUserId);
```
Comment (alshakero) [inline src/Messages/Messages.js:116]:
```suggestion

```

File: src/Messages/Messages.module.css [hunk 7]
```diff
@@ -0,0 +1,191 @@
+.messagesPageContainer {
+  max-width: 935px;
+  margin: 80px auto;
+  display: flex;
+}
```
Comment (alshakero) [inline src/Messages/Messages.module.css:5]: This needs a border radius. Check IG for reference.

File: src/Messages/Messages.module.css [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+  justify-content: space-between;
+  align-items: center;
+  background-color: #fff;
+}
+.sendMessageIcon {
+  padding-right: 20px;
+}
+.usernameIconSection {
+  display: flex;
+  margin: 30px 20px 30px 120px;
```
Comment (alshakero) [inline src/Messages/Messages.module.css:25]: I appreciate the creativity, but this not good. Writing the coordinates manually makes your CSS brittle (it breaks easily). I would add an empty `div` in `userSection`. The give `flex: 1` to everything inside `userSection`. This way you'll have 3 equal parts with the username in the center. 

```css
userSection > * {
	flex: 1
}
```

File: src/Messages/Messages.module.css [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+.arrowDownIcon {
+  transform: rotate(180deg);
+  align-self: center;
+  padding: 0 10px;
+}
+.sendersSection {
+  width: 350px;
+  border: 1px solid #dbdbdb;
+  overflow: hidden auto;
+  height: 50%;
```
Comment (alshakero) [inline src/Messages/Messages.module.css:37]: hmmmm. 50% filling the viewport means the container (`leftSection`) is too big. That's why you have empty scrolling below on the screen. Make this 100%, then make the container smaller. I'd make the container of this div like this

```css
.leftSection {
	/* screen height - margin top - margin bottom */
	height: calc( 100vh - 80px - 80px); 
}
```

If you do this for both sides (left and right) you'll get the same responsiveness as IG.

File: src/Messages/Messages.js [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+  messagesSection,
+  messagesSenderAvatar,
+  messagesSenderInfo as messagesHeader,
+  messagesSenderUserInfo,
+  messagesSenderUsername,
+  rightSection,
+  sendAMessage,
+  senderInfo,
+  sendersSection,
+  SenderThumbnail,
```
Comment (alshakero) [inline src/Messages/Messages.js:26]:
```suggestion
  senderThumbnail,
```

File: src/Messages/Messages.module.css [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+.sendersSection::-webkit-scrollbar {
+  display: none;
+}
+.senderInfo {
+  display: flex;
+  padding: 8px 20px;
+  gap: 10px;
+  align-items: center;
+}
+.SenderThumbnail {
```
Comment (alshakero) [inline src/Messages/Messages.module.css:49]:
```suggestion
.senderThumbnail {
```

File: src/Messages/Messages.module.css [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+  padding-right: 8px;
+  border: 1px solid #dbdbdb;
+  border-radius: 22px;
+  align-items: center;
+}
+.messagesInputSection {
+  padding: 20px;
+}
+.messagesInput {
+  height: 18px !important;
```
Comment (alshakero) [inline src/Messages/Messages.module.css:140]: I don't think you need this.
```suggestion
  height: 18px;
```

File: src/Messages/Messages.module.css [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+}
+.telegramIconContainer {
+  width: 96px;
+  height: 96px;
+  border: 3px solid black;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  border-radius: 50%;
```
Comment (alshakero) [inline src/Messages/Messages.module.css:168]: The icon isn't centered inside this. Maybe some padding-top of 10px and box-sizing obviously.

**Review by alshakero — CHANGES_REQUESTED** [review]
Loving it! Left some comments. I'll leave some here as well:

1. The message entry should have a background when it's active. See here 
[here](https://user-images.githubusercontent.com/17054134/143139204-cf2376e5-8e13-4eaa-882f-54613259346e.png).
2. The whole message entry should be clickable, not only the photo.

File: src/Messages/Messages.module.css (hunk 9 above)
Comment (alshakerM) [reply src/Messages/Messages.module.css:37]: On it

File: src/Icons/Icons.js [hunk 14]
```diff
@@ -1,33 +1,48 @@
...
+          aria-label="Direct"
+          className="icon "
+          color="#262626"
+          fill="#262626"
+          height="22"
+          role="img"
+          viewBox="0 0 48 48"
+          width="22"
+        >
+          {location.pathname === '/direct/inbox' || location.pathname === `/direct/t/${from_user_id}`? (
```
[the remark this answers is not in the document]
Comment (alshakerM) [reply src/Icons/Icons.js:39]: Didn't know there was a startsWith 🗡️ thanks for the info

**Comment by alshakerM** [comment]
the title says it all 👯

**Comment by alshakerM** [comment]
Changed the if statement in icons to startsWith instead of || for cleaner and DRYer code.

**Comment by alshakerM** [comment]
Deleted some unused CSS and edit the height of the senderSection and messagesSection.

**Comment by alshakerM** [comment]
Added some padding to the main telegram icon and made it a bit larger to look like the real IG.
Edited a className to be camelCase.
Removed !important from messagesInput.

**Comment by alshakerM** [comment]
Deleted the CSS file and moved its content to the module CSS file.
Remove height from the messagesSection.
Merged Routes in App.js.

**Comment by alshakerM** [comment]
Changed the time in senderSection.
Removed console log.
Added border radius ( not the main div but to each corner section to avoid using overflow ).
Removed margin from the icon in userSection and added an empty div.

File: src/Messages/Messages.js [hunk 15]
```diff
@@ -0,0 +1,285 @@
...
+  sendMessageIcon,
+  sendMessagesSection,
+  telegramIcon,
+  telegramIconContainer,
+  usernameIconSection,
+  userSection,
+  whenWasActive,
+  yourMessages,
+  emptyUserName,
+} from './Messages.module.css';
```
Comment (alshakero) [inline src/Messages/Messages.js:41]: In the future you can import all these like so:

```js
import styles from './Messages.module.css'
```

then 
```jsx
<div className={styles.messageText} ...
```

File: src/Messages/Messages.js [hunk 16]
```diff
@@ -0,0 +1,285 @@
...
+    <div className={sendersSection}>
+      {messagesData.map((user) => (
+        <Link
+          to={`/direct/t/${user.from_user_id}`}
+          style={
+            user.from_user_id === fromUserId
+              ? { backgroundColor: '#efefef' }
+              : {}
+          }
+          className={senderInfo}
```
Comment (alshakero) [inline src/Messages/Messages.js:92]: We can let this slip. But it's better to create `isSelected` class and use classnames.

File: src/Messages/Messages.js [hunk 17]
```diff
@@ -0,0 +1,285 @@
...
+              />
+              <div className={messagesSenderUserInfo}>
+                <p className={messagesSenderUsername}>
+                  <strong>{messagesData[userIndex]?.from_username}</strong>
+                </p>
+                <p className={whenWasActive}>Active 20min ago</p>
+              </div>
+              <svg
+                aria-label="View thread details"
+                class="_8-yf5 "
```
Comment (alshakero) [inline src/Messages/Messages.js:159]: Unneeded class.

File: src/Messages/Messages.js [hunk 18]
```diff
@@ -0,0 +1,285 @@
...
+                      </p>
+                    </div>
+                  );
+                })}
+              </div>
+              <div className={messagesInputSection}>
+                <div className={messagesInputContent}>
+                  <svg
+                    aria-label="Emoji"
+                    class="_8-yf5 "
```
Comment (alshakero) [inline src/Messages/Messages.js:228]: Unneeded class.

File: src/Messages/Messages.js [hunk 19]
```diff
@@ -0,0 +1,285 @@
...
+                    <path d="M34.9 24c0-1.4-1.1-2.5-2.5-2.5s-2.5 1.1-2.5 2.5 1.1 2.5 2.5 2.5 2.5-1.1 2.5-2.5zm-21.8 0c0-1.4 1.1-2.5 2.5-2.5s2.5 1.1 2.5 2.5-1.1 2.5-2.5 2.5-2.5-1.1-2.5-2.5zM24 37.3c-5.2 0-8-3.5-8.2-3.7-.5-.6-.4-1.6.2-2.1.6-.5 1.6-.4 2.1.2.1.1 2.1 2.5 5.8 2.5 3.7 0 5.8-2.5 5.8-2.5.5-.6 1.5-.7 2.1-.2.6.5.7 1.5.2 2.1 0 .2-2.8 3.7-8 3.7z"></path>
+                  </svg>
+                  <input
+                    type="text"
+                    placeholder="Message..."
+                    className={messagesInput}
+                  />
+                  <svg
+                    aria-label="Add Photo or Video"
+                    class="_8-yf5 "
```
Comment (alshakero) [inline src/Messages/Messages.js:246]: +1

File: src/NavBar/NavBar.js [hunk 20]
```diff
@@ -97,7 +97,7 @@ export function NavBar() {
           </BorderLessTextInput>
         </div>
         <div className="nav-icons">
-          <Icons />
+          <Icons from_user_id={from_user_id}/>
```
Comment (alshakero) [inline src/NavBar/NavBar.js:100]:
```suggestion
          <Icons fromUserId={from_user_id}/>
```

**Review by alshakero — APPROVED** [review]
Looks amazing. Left some cosmetic remarks. Feel free to merge after addressing them.

**Comment by alshakero** [comment]
"Final changes" isn't a great commit message, given you merged it to master. Please either squash your PRs or pay attention to commit messages.

**Pull request merged**